source/georef/street_network.cpp

```diff
@@ -237,21 +236,121 @@ void PathFinder::start_distance_dijkstra(const navitia::time_duration& radius) {
 
 }
 
+void PathFinder::start_target_all_dijkstra(const std::vector<vertex_t>& targets) {
+    if (! starting_edge.found)
+        return ;
+    computation_launch = true;
+    try {
+        dijkstra(starting_edge[source_e], target_all_visitor{targets});
+    } catch(const DestinationFound&){}
+
+    try {
+        dijkstra(starting_edge[target_e], target_all_visitor{targets});
+    } catch(const DestinationFound&){}
+}
+
 std::vector<std::pair<type::idx_t, type::GeographicalCoord>>
 PathFinder::crow_fly_find_nearest_stop_points(const navitia::time_duration& radius,
                                               const proximitylist::ProximityList<type::idx_t>& pl) {
     // Searching for all the elements that are less than radius meters awyway in crow fly
     float crow_fly_dist = radius.total_seconds() * speed_factor * georef::default_speed[mode];
-
     return pl.find_within(start_coord, crow_fly_dist);
 }
 
+
+struct ProjectionGetterByCache{
+    const nt::Mode_e mode;
+    const std::vector<GeoRef::ProjectionByMode>& projection_cache;
+    const georef::ProjectionData& operator()(const routing::SpIdx& idx) const{
+        return projection_cache[idx.val][mode];
+    }
+};
+
+struct ProjectionGetterOnFly{
+    const GeoRef& geo_ref;
+    const nt::idx_t& offset;
+    const georef::ProjectionData operator()(const type::GeographicalCoord& coord) const{
+        return georef::ProjectionData{coord, geo_ref, offset, geo_ref.pl};
+    }
+};
+
+static routing::SpIdx get_id(const routing::SpIdx& idx) { return idx; }
+static std::string get_id(const type::GeographicalCoord& coord) { return coord.uri(); }
+
+template<typename K, typename U, typename G>
+boost::container::flat_map<K, navitia::time_duration>
+PathFinder::start_dijkstra_and_fill_duration_map(const navitia::time_duration& radius,
+        const std::vector<U>& destinations,
+        const G& projection_getter,
+        Dijkastra_Type dijkastra_type){
```

Comment:
@kinnou02 After a short discussion with @TeXitoi, I realized that we do not have to use the target_all_visitor for stopping the dijkastra, we only need the distance_visitor since we did a crow fly research before launching the dijkastra.